### PR TITLE
Ducc temp directory

### PR DESCRIPTION
**Pull request by siscia**
This PR add the possibility to specify a temporary directory that DUCC can use in lieu of /tmp.

There are some other minor changes as well addressing: https://sft.its.cern.ch/jira/projects/CVM/issues/CVM-1840

### CODE REVIEW

File: test/src/401-ducc-convert_singularity/main [hunk 1]
```diff
@@ -4,10 +4,15 @@ cvmfs_test_suites="ducc"
...
+ducc_test_401_clean_up() {
+  sudo cvmfs_server rmfs -f $CVMFS_TEST_REPO
+}
 
 cvmfs_run_test() {
+  trap ducc_test_401_clean_up EXIT HUP INT TERM
+
   echo "*** creating empty repo ***"
-  create_empty_repo $CVMFS_TEST_REPO $CVMFS_TEST_USER || return 1
+  create_empty_repo $CVMFS_TEST_REPO $USER || return 1
```
Comment (jblomer) [inline test/src/401-ducc-convert_singularity/main]: Why `$USER` instead of `$CVMFS_TEST_USER`

Comment (siscia) [reply test/src/401-ducc-convert_singularity/main]: DUCC expect to be run by the user who owns the repository.

Comment (jblomer) [reply test/src/401-ducc-convert_singularity/main]: I see, thanks

**Comment by siscia** [comment]
I will wait tomorrow to sleep on it and then I will merge.

**Comment by DrDaveD** [comment]
Are you OK with the fact that --fix-perms implicitly requires singularity >= 3.5.0? 

Maybe the rpm should say `Conflicts: singularity < 3.5.0`.

**Comment by siscia** [comment]
Yeah it is going to be fixed in the next PR.

And indeed I should fix the rpm as well! Thanks I didn't think about it!

**Comment by siscia** [comment]
Also, this address, even if under the radar: [CVM-1832](https://sft.its.cern.ch/jira/projects/CVM/issues/CVM-1832)

**Pull request merged**